packages/jest-cli/src/watch.js

```diff
@@ -68,19 +72,71 @@ export default function watch(
   });
 
   const updateConfigAndRun = ({
+    bail,
+    collectCoverage,
+    collectCoverageFrom,
+    collectCoverageOnlyFrom,
+    coverageDirectory,
+    coverageReporters,
     mode,
+    notify,
+    notifyMode,
+    onlyFailures,
+    reporters,
     testNamePattern,
     testPathPattern,
     updateSnapshot,
+    verbose,
   }: {
+    bail?: boolean,
+    collectCoverage?: boolean,
+    collectCoverageFrom?: Array<string>,
+    collectCoverageOnlyFrom?: {[key: string]: boolean},
+    coverageDirectory?: string,
+    coverageReporters?: Array<string>,
     mode?: 'watch' | 'watchAll',
+    notify?: boolean,
+    notifyMode?: string,
+    onlyFailures?: boolean,
+    reporters?: Array<ReporterConfig>,
     testNamePattern?: string,
     testPathPattern?: string,
     updateSnapshot?: SnapshotUpdateState,
+    verbose?: boolean,
   } = {}) => {
     const previousUpdateSnapshot = globalConfig.updateSnapshot;
+    // It is unfortunate that Flow isn't smart enough yet to let us
+    // implement a generic assign loop over a list of white-listed
+    // option names here :-(
     globalConfig = updateGlobalConfig(globalConfig, {
```

Comment:
Now that there are a lot of options, maybe we could move a lot of this `!== undefined` logic into the `updateGlobalConfig` function, so that it is the only place that knows how to handle all of that logic

Reply:
Yeah, that's an option. I really disliked the un-DRYness of this part of the code, but making it dynamically iterate over a list of props confused the heck out of Flow 😒 .

I'll try and move that logic inside `updateGlobalConfig` and see how it goes.